Dockerfile

```diff
@@ -1,72 +1,79 @@
-FROM ubuntu:20.04 as wgrib2
-
-WORKDIR /src
-
-ENV WGRIB2_VERSION="v2.0.8"
-
-# Update the repo
-RUN apt-get update && \
-    apt-get install -y build-essential libaec-dev zlib1g-dev libcurl4-openssl-dev libboost-dev curl wget zip unzip bzip2 gfortran gcc g++
-
-
-# Download the latest wgrib2 source code
-RUN wget -c ftp://ftp.cpc.ncep.noaa.gov/wd51we/wgrib2/wgrib2.tgz.$WGRIB2_VERSION && \
-    tar -xzvf wgrib2.tgz.$WGRIB2_VERSION && \
-    cd grib2 && \
-    CC=gcc FC=gfortran make
-
-# Build the virtual environment in an isolated container
-FROM python:3.11 as builder
-
-RUN pip install poetry==1.8.2
-
+# Build the reqired depencies
+FROM continuumio/miniconda3 as builder
+
+# Install and package up the conda environment
+# Creates a standalone environment in /opt/venv
+COPY environment.yml /opt/environment.yml
+RUN conda env create -f /opt/environment.yml
+RUN conda install -c conda-forge conda-pack poetry=1.8.2
+RUN conda-pack -n setup_wrf -o /tmp/env.tar && \
+  mkdir /opt/venv && cd /opt/venv && \
+  tar xf /tmp/env.tar && \
+  rm /tmp/env.tar
+
+# We've put venv in same path it'll be in final image,
+# so now fix up paths:
+RUN /opt/venv/bin/conda-unpack
+
+# Install the python dependencies using poetry
 ENV POETRY_NO_INTERACTION=1 \
-    POETRY_VIRTUALENVS_IN_PROJECT=1 \
-    POETRY_VIRTUALENVS_CREATE=1 \
+    POETRY_HOME='/opt/venv' \
+    POETRY_VIRTUALENVS_CREATE=false \
     POETRY_CACHE_DIR=/tmp/poetry_cache
 
-WORKDIR /app
+# This is deliberately outside of the work directory
+# so that the local directory can be mounted as a volume of testing
+ENV VIRTUAL_ENV=/opt/venv \
+    PATH="/opt/venv/bin:$PATH"
+
+# Needed for wgrib2
+RUN ln -s /opt/venv/lib/libnetcdf.so /opt/venv/lib/libnetcdf.so.13
```

Comment:
@mikapfl Sorry 😬 I have no idea why wgrib2 targets such an old version of libnetcdf. This is a very hacky fix, but better than continuing to build wgrib2 from source

Reply:
Well, if it works, it works… Otherwise, it segfaults, I guess.